.github/workflows/pr-scan.yml

```diff
@@ -0,0 +1,87 @@
+name: PR Security Scan
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: 'recursive'
+    - name: Build the Docker image
+      uses: docker/build-push-action@v5
+      with: 
+        context: .
+        push: false
+        load: true
+        tags: pr_image:${{ github.sha }}
+    - name: Run Trivy vulnerability scanner on image
+      uses: aquasecurity/trivy-action@master
```

Comment:
Best practice is to pin a version so let's use a version here instead of `master`